x-pack/plugins/security_solution/server/endpoint/routes/artifacts/download_exception_list.test.ts

```diff
@@ -0,0 +1,301 @@
+      method: 'get',
+      params: { sha256: '789' },
+      headers: {
+        authorization: AuthHeader,
+      },
+    });
+
+    ingestSavedObjectClient.get.mockImplementationOnce(() =>
+      // eslint-disable-next-line prefer-promise-reject-errors
+      Promise.reject({ output: { statusCode: 404 } })
+    );
+
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/allowlist/download')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          savedObjects: {
+            client: mockSavedObjectClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+    expect(mockResponse.notFound).toBeCalled();
+  });
+
+  it('should utilize the cache', async () => {
+    const mockSha = '123456789';
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: `/api/endpoint/allowlist/download/${mockArtifactName}/${mockSha}`,
+      method: 'get',
+      params: { sha256: mockSha, identifier: mockArtifactName },
+      headers: {
+        authorization: AuthHeader,
+      },
+    });
+
+    // Add to the download cache
+    const mockCompressedArtifact = await compressExceptionList(expectedEndpointExceptions);
+    const cacheKey = `${mockArtifactName}-${mockSha}`;
+    cache.set(cacheKey, mockCompressedArtifact.toString('binary'));
+
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/allowlist/download')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          savedObjects: {
+            client: mockSavedObjectClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+    expect(mockResponse.ok).toBeCalled();
+    // The saved objects client should be bypassed as the cache will contain the download
+    expect(ingestSavedObjectClient.get.mock.calls.length).toEqual(0);
+  });
+
+  it('should respond with a 401 if a valid API Token is not supplied', async () => {
+    const mockSha = '123456789';
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: `/api/endpoint/allowlist/download/${mockArtifactName}/${mockSha}`,
+      method: 'get',
+      params: { sha256: mockSha, identifier: mockArtifactName },
+    });
+
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/allowlist/download')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          savedObjects: {
+            client: mockSavedObjectClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+    expect(mockResponse.unauthorized).toBeCalled();
+  });
+
+  it('should respond with a 404 if an agent cannot be linked to the API token', async () => {
```

Comment:
I think this should probably be a 401. 404 could be appropriate though, if it's an authorization issue (as per W3C guidelines)... we should talk through this.